gril/grilreply.c

```diff
@@ -493,15 +493,29 @@ struct reply_data_reg_state *g_ril_reply_parse_data_reg_state(GRil *gril,
 
 	g_ril_append_print_buf(gril, "{");
 
+
+        /* DANGER WILL ROBINSON
+         * In some cases from Vodaphone we are receiving a RAT of 102
+         * while in tunnels of the metro. Let's assume that if we
+         * receive 102 we actually want a RAT of 2 for EDGE service */
+/*	if (g_ril_vendor(gril) == OFONO_RIL_VENDOR_SAMSUNG_MSM_822x) {
+                if (str_arr->num_str > 4 && strcmp(str_arr->str[0], "1") && strcmp(str_arr->str[3], "102")) {
+			str_arr->str[3] = "2";
+		}
+	}
+*/
+
```

Comment:
Is this still needed?
- If yes, I'm not sure if the assignment at line 503 works. I thought an `strcpy()` or manual `NULL` insertion is required.
- If no, please remove it.

Reply:
To be honest I have no idea if it is needed. Its in the original Java file like this. Line 503 probably does not work, but I will remove it entirely. Its also probably not a device-specific thing at all, they just found this and made a drive-by fix.